src/Engine/Manager/RenderManager.cpp

```diff
@@ -259,45 +280,77 @@ void RenderManager::Render(World& world, const glm::mat4& translationMatrix, con
             }
         }
     }
+    }
+    }
 
     /// @todo Render skinned meshes.
     
     // Light the world.
-    {
-        PROFILE("Light the world");
+    { PROFILE("Light the world");
+    { GPUPROFILE("Light the world", Video::Query::Type::TIME_ELAPSED);
+    { //GPUPROFILE("Light the world", Video::Query::Type::SAMPLES_PASSED);
         LightWorld(world, viewMatrix, projectionMatrix, viewProjectionMatrix, renderSurface);
     }
+    }
+    }
+
     /*
     // Anti-aliasing.
     if (Hymn().filterSettings.fxaa) {
-        PROFILE("Anti-aliasing(FXAA)");
-        renderer->AntiAlias(renderSurface);
+        { PROFILE("Anti-aliasing(FXAA)");
+        { GPUPROFILE("Anti-aliasing(FXAA)", Video::Query::Type::TIME_ELAPSED);
+        { GPUPROFILE("Anti-aliasing(FXAA)", Video::Query::Type::SAMPLES_PASSED);
+            renderer->AntiAlias(renderSurface);
+        }
+        }
+        }
     }
-
+    
+    
     // Fog.
     if (Hymn().filterSettings.fog) {
-        PROFILE("Fog");
-        renderer->RenderFog(renderSurface, projectionMatrix, Hymn().filterSettings.fogDensity, Hymn().filterSettings.fogColor);
+        { PROFILE("Fog");
+        { GPUPROFILE("Fog", Video::Query::Type::TIME_ELAPSED);
+        { GPUPROFILE("Fog", Video::Query::Type::SAMPLES_PASSED);
+            renderer->RenderFog(renderSurface, projectionMatrix, Hymn().filterSettings.fogDensity, Hymn().filterSettings.fogColor);
+        }
+        }
+        }
     }
-
+    
     // Render particles.
-    {
-        PROFILE("Render particles");
+    { PROFILE("Render particles");
+    { GPUPROFILE("Render particles", Video::Query::Type::TIME_ELAPSED);
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') The function 'GetAnimations' is never used. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)